src/pretix/base/services/orders.py

```diff
@@ -515,6 +515,42 @@ def send_expiry_warnings(sender, **kwargs):
             else:
                 o.log_action('pretix.event.order.expire_warning_sent')
 
+@receiver(signal=periodic_task)
+def send_download_reminders(sender, **kwargs):
+    eventcache = {}
+    today = now().replace(hour=0, minute=0, second=0)
+
+    for o in Order.objects.filter(expires__gte=today + timdelta(days=5), download_reminder_sent=False,
```

Comment:
Sure. Sorry that it took me so long to reply.

What this currently does:

* Find all orders which
  * expire in the next five days and
  * have not been sent a reminder yet and
  * are in status pending
* For each of these orders
  * Look into the event settings for the configured number of days N (i.e. 3)
  * Check if the order expires in the next N days
  * If so, send an email

Here, we need to loop over a different set of orders and check different things. To make it easier, you can split it into two for loops and throw out all that ``eventcache`` stuff in there. It coult then look like:

* Find all events which
  * happen in the future (``date_from__gte=today``)
* For each of these events
  * Check if ``mail_days_download_reminder`` is not None, ``continue`` to skip the event otherwise
  * Calculate the date for reminders (``reminder_date = (event.date_from - timedelta(days=event.settings.mail_days_download_reminder)).replace(hour=0, minute=0, second=0)``)
  * Check if that date is already reached (``today > reminder_date``). If not, ``continue`` to skip this event
  * Find all orders which
    * Belong to this event (``event.orders.filter(…)``)
    * Have the status paid
    * Have not received the reminder email yet
  * For each of these orders
     * Send the email

Does that make it more clear?